src/main/kotlin/org/rust/ide/utils/RustModTabTitleProvider.kt

```diff
@@ -0,0 +1,18 @@
+package org.rust.ide.utils
+
+import com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import org.rust.lang.core.psi.impl.isRustFile
+
+class RustModTabTitleProvider : EditorTabTitleProvider {
+    val MOD_FILE_NAME = "mod.rs"
+
+    override fun getEditorTabTitle(project: Project, file: VirtualFile): String? {
+        if (file.isRustFile && MOD_FILE_NAME == file.name) {
+            return "${file.parent.name}/${file.name}"
```

Comment:
Yep, it is helpful to prepend a directory name even if a single `mod.rs` is opened. However, when several `mod.rs` files are open, we should fall back to the default IDEA behavior.